drivers/interrupt_controller/arcv2_irq_unit.c

```diff
@@ -162,8 +162,11 @@ static int _arc_v2_irq_unit_resume(struct device *dev)
 		_arc_v2_aux_reg_write(_ARC_V2_IRQ_ENABLE,
 				ctx.irq_config[irq - 16] & BIT(0));
 	}
-
+#ifdef CONFIG_ARC_NORMAL_FIRMWARE
+	/* \todo use sjli instruction to access irq_ctrl */
```

Comment:
maybe suspend/resume functionality should have different overall implementation, Secure firmware should probably also support suspend/resume for this to work.